api/location/locationRouter.js

```diff
@@ -0,0 +1,76 @@
+const express = require('express');
+const DB = require('../utils/db-helper');
+const router = express.Router();
+const { requireAdmin } = require('../middleware/authorization');
+
+// GET - View all locations
+router.get('/', (req, res) => {
+  DB.findAll('locations')
+    .then((locations) => {
+      res.status(200).json(locations);
+    })
+    .catch((err) => {
+      res.status(500).json({ error: err.message });
+    });
+});
+
+// GET - View Location by ID
+router.get('/:id', (req, res) => {
+  const { id } = req.params;
+
+  DB.findById('locations', id)
+    .then((Location) => {
+      if (Location) {
+        res.status(200).json(Location);
+      } else {
+        res.status(404).json({ error: `Location ${id} not found` });
+      }
+    })
+    .catch((err) => {
+      res.status(500).json({ error: err.message });
+    });
+});
+
+// POST - Create new Location
+router.post('/', (req, res) => {
+  DB.create('locations', req.body)
+    .then((newLocation) => {
+      res.status(201).json({ message: 'Location created', newLocation });
+    })
+    .catch((err) => {
+      res.status(500).json({ error: err.message });
+    });
+});
+
+// PUT - Update Location by ID
+router.put('/:id', (req, res) => {
+  DB.update('locations', req.params.id, req.body)
+    .then((editedLocation) => {
+      res.status(200).json({
+        message: `Location ${req.params.id} updated`,
+        editedLocation,
+      });
+    })
+    .catch((err) => {
+      res.status(500).json({ error: err.message });
+    });
+});
+
+// DELETE - Remove Location by ID - *only Admins can delete locations
+router.delete('/:id', requireAdmin, (req, res) => {
```

Comment:
Good job with implementing authorization middleware!

I'm not 100% sure I agree with your decision to limit the ability though. Since the table is "RESTRICT" by default, it would only be able to be deleted if nothing were referencing the location key.  However if the service provider made a typo, and no one was using that location yet, they could go in and delete/edit it. That's my thinking at least.

If you do decide to keep the middleware in, I think Program Managers should be allowed to delete as well. This would require a new authorization function, one that allows both Administrators and Program Managers.

Reply:
The Router looks solid.  I agree that Program Managers should be given access to this endpoint, but I'd leave the middleware in for now.  This sounds like a great question to ask the stakeholder at the next meeting!